### DISCUSSION TITLE
In TemporaryMemoryManager what is the significance of 512 blocks?

### DISCUSSION THREAD

**Discussion by pmg1991** [duckdb/duckdb#12187] [Q&A]
For TemporaryMemoryManager this is the current hard-coded reservation

`MINIMUM_RESERVATION_PER_STATE_PER_THREAD = idx_t(512) * Storage::BLOCK_ALLOC_SIZE;`

What if instead of 512 a lower or higher value is used? 
Can any one explain what impact it creates or the reasoning behind this?

**Comment by Tishj** [comment]
Usually a good place to look is using git blame to find the PR that introduced it
<https://github.com/duckdb/duckdb/pull/10147>

Though I have to say, it doesn't come up there

**Reply by pmg1991** [reply]
Yes, I've read it , but it didn't help much in this case.

**Reply by Tishj** [reply]
```c++
	//! TemporaryMemoryState is initialized with a minimum reservation guarantee, which is either
	//! 512 blocks per state per thread, which is 0.125GB per thread for Storage::BLOCK_ALLOC_SIZE = 262144
	static constexpr const idx_t MINIMUM_RESERVATION_PER_STATE_PER_THREAD = idx_t(512) * Storage::BLOCK_ALLOC_SIZE;
```
	
the reasoning is likely to just pick a sane minimum, acquired through testing
